### DISCUSSION TITLE
Does checkout supports sparse-checkout

### DISCUSSION THREAD

**Discussion by fawazahmed0** [community/community#26578] [Actions]
<p>This will help in case of large repos, just wanted to know whether this is supported or not in checkout actions<br>
We can combine this with clone --filter flag to clone only the files we want</p>

**Comment by kingthorin** [comment, marked as answer]
<p>You can specify <code>fetch-depth</code>, which might help:<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/actions/checkout" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/7570458/181098544-021a7c5d-8316-48b1-9c31-e6881c4066d9.png" class="thumbnail onebox-avatar" width="400" height="400">

<h3><a href="https://github.com/actions/checkout" target="_blank" rel="nofollow noopener">actions/checkout</a></h3>

<p>Action for checking out a repo. Contribute to actions/checkout development by creating an account on GitHub.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>It looks like there’s an open issue to support <code>sparse</code> checkout:<br>
</p><aside class="onebox githubissue">
  <header class="source">
      <a href="https://github.com/actions/checkout/issues/172" target="_blank" rel="nofollow noopener">github.com/actions/checkout</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/actions/checkout/issues/172" target="_blank" rel="nofollow noopener">Options like sparse mode</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2020-03-02" data-time="12:57:52" data-timezone="UTC">12:57PM - 02 Mar 20 UTC</span>
      </div>


      <div class="user">
        <a href="https://github.com/jjangga0214" target="_blank" rel="nofollow noopener">
          <img alt="jjangga0214" src="https://user-images.githubusercontent.com/7570458/181098545-108d8f69-781e-47b0-bb1d-0e24fbfeced0.png" class="onebox-avatar-inline" width="20" height="20">
          jjangga0214
        </a>
      </div>
    </div>
  </div>
</div>

<div class="github-row">
  <p class="github-content">Hi.
GitHub is internally evaluating partial-clone (Actually GitLab already provides the feature!).
This can be useful for a large monorepo with sparse-checkout(see git...</p>
</div>

<div class="labels">
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by fawazahmed0** [comment]
<p>I was just thinking of opening a new feature request , Thanks for point to the already open issue.</p>

**Comment by fawazahmed0** [comment]
<p>This is what I did to get partial clone:</p>
<pre><code class="lang-yaml">      - name: Partial Clone
        run: |
          REPO="https://${GITHUB_ACTOR}:${{ secrets.GITHUB_TOKEN }}@github.com/${GITHUB_REPOSITORY}.git"
          git config --global user.email github-actions@github.com
          git config --global user.name github-actions
          git clone --filter=blob:none --no-checkout --depth 1  --sparse $REPO .
          git sparse-checkout init --cone
          git sparse-checkout add "folder1" "folder2/folder3"
          git checkout
</code></pre>
<p>Hopefully should help someone trying to clone large repos</p>
<p>Refer <a href="https://stackoverflow.com/a/63786181/2437224" rel="nofollow noopener">this</a> and <a href="https://git-scm.com/docs/git-sparse-checkout" rel="nofollow noopener">this</a> for more details</p>

**Comment by steoj** [comment]
<p>This looks good, but I have not been able to decode the GITHUB_ACTOR and GITHUB_TOKEN. Are these personal or something to explicitly allow for my private repo?</p>

**Comment by kingthorin** [comment]
<p>GITHUB_ACTOR is whoever is causing the action to run. (Ref: <a href="https://docs.github.com/en/actions/reference/environment-variables" class="inline-onebox" rel="noopener nofollow ugc">Environment variables - GitHub Docs</a>)</p>
<p>GITHUB_TOKEN is a temporary token that’s generated for each action run. (Ref: <a href="https://docs.github.com/en/actions/reference/authentication-in-a-workflow" class="inline-onebox" rel="noopener nofollow ugc">Authentication in a workflow - GitHub Docs</a>)</p>
